cpp/src/render/2d/cuda/heatmap.cu

```diff
@@ -173,14 +173,44 @@ void set_colors_gpu(float *colors,
 } //namespace render
 } //namespace
 
+#define TEMPLATE_GEN_PREFIX
```

Comment:
why multiple '#define TEMPLATE_GEN_PREFIX'??
do you mean #ifdefine TEMPLATE_GEN_PREFIX'?